Template.php

```diff
@@ -832,7 +832,7 @@ protected function get_identifiers_from_url($url_sent = NULL) {
         }
         return $this->add_if_new("citeseerx", urldecode($match[1])); // We cannot parse these at this time
         
-      } elseif (preg_match("~\barxiv\.org/.*(?:pdf|abs)/(.+)$~", $url, $match)) {
+      } elseif (preg_match("~\barxiv\.org/.*(?:pdf|abs|tp/arxiv/papers/\d{4})/(.+?)(?:\.pdf)?$~us", $url, $match)) {
```

Comment:
Do we need unicode support for URLs?
and have you deliberately omitted the f from ftp?